pkg/client/unversioned/helper.go

```diff
@@ -428,7 +440,13 @@ func UnversionedRESTClientFor(config *Config) (*RESTClient, error) {
 		return nil, err
 	}
 
-	client := NewRESTClient(baseURL, unversioned.SchemeGroupVersion, config.Codec, config.QPS, config.Burst)
+	versionConfig := config.ContentConfig
+	if versionConfig.GroupVersion == nil {
+		v := unversioned.SchemeGroupVersion
+		versionConfig.GroupVersion = &v
```

Comment:
Only if `config.ContentConfig.GroupVersion` is `nil`. What I was saying is this is a new behavior allowing a for a different `GroupVersion` value than `unversion.SchemeGroupVersion`